src/turnkeyml/common/build.py

```diff
@@ -137,12 +137,25 @@ def hash_model(model, model_type: ModelType, hash_params: bool = True):
         raise ValueError(msg)
 
 
-class Status(enum.Enum):
+class FunctionStatus(enum.Enum):
+    # INCOMPLETE indicates stage/build/benchmark is either running or was killed;
+    # if you know the process ended then it was killed;
+    # if the process is still running, stage/build/benchmark is still running.
+    INCOMPLETE = "incomplete"
+    # NOT_STARTED applies to stages that didnt start because
+    # the build errored out or was killed prior to stage starting.
     NOT_STARTED = "not_started"
-    PARTIAL_BUILD = "partial_build"
-    BUILD_RUNNING = "build_running"
-    SUCCESSFUL_BUILD = "successful_build"
-    FAILED_BUILD = "failed_build"
+    # SUCCESSFUL means the stage/build/benchmark completed successfully
+    SUCCESSFUL = "successful"
+    # ERROR means the stage/build/benchmark failed and threw some error that
+    # was caught by turnkey. You should proceed by looking at the build
+    # logs to see what happened.
+    ERROR = "error"
+    # KILLED means the build/benchmark failed because the system killed it. This can
+    # happen because of an out-of-memory (OOM), timeout, system shutdown, etc.
+    # You should proceed by re-running the build and keeping an eye on it to observe
+    # why it is being killed (e.g., watch the RAM utilization to diagnose an OOM).
+    KILLED = "killed"
```

Comment:
I previously suggested changes to the release notes that included this description. Please make sure to change both.